docker/lib/dependabot/docker/file_fetcher.rb

```diff
@@ -40,63 +42,72 @@ def fetch_files
         elsif incorrectly_encoded_dockerfiles.none?
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_yamlfiles.first.path
+            T.must(incorrectly_encoded_yamlfiles.first).path
           )
         else
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_dockerfiles.first.path
+            T.must(incorrectly_encoded_dockerfiles.first).path
           )
         end
       end
 
       private
 
+      sig { returns(T::Array[DependencyFile]) }
       def dockerfiles
-        @dockerfiles ||=
+        @dockerfiles ||= T.let([], T.nilable(T::Array[T.untyped]))
+        @dockerfiles +=
```

Comment:
Won't this now always append to the array, since the previous line sets it to an empty array if unset, but next time the method is called it will not set it to nil, but still execute this line, appending to the array?

Reply:
I believe you can use the following pattern instead:

```ruby
sig { returns(Foo) }
def foo
  @foo ||= T.let(Foo.new, T.nilable(Foo))
end

```

Reply:
Yeah, you should be able to wrap the whole `||=` statement in `T.let`, and not have to add the `+=` statement.

Reply:
This is fixed as per review comments